File: src/lightnap-ng/src/app/core/backend-api/dtos/users/request/public-search-users-request-dto.ts
import { PagedRequestDto } from "../../paged-request-dto";
import { SearchUsersSortBy } from "./search-users-sort-by";

/**
 * Interface representing a request to search for users.
 * Extends the PaginationRequest interface to include pagination properties.
 */
export interface PublicSearchUsersRequestDto extends PagedRequestDto {
    /**
     * The username substring to search for.
     * @type {string}
     */
    userName?: string;

    /**
     * The field to sort the results by.
     * @type {SearchUsersSortBy}
     */
    sortBy: SearchUsersSortBy;

    /**
     * Whether to reverse the sort order.
     * @type {boolean}
     */
    reverseSort: boolean;
}

File: src/LightNap.Core/Users/Dto/Request/PublicSearchUsersRequestDto.cs
﻿using LightNap.Core.Api;
using LightNap.Core.Users.Models;

namespace LightNap.Core.Users.Dto.Request
{
    /// <summary>
    /// Represents a request to search users.
    /// </summary>
    public class PublicSearchUsersRequestDto : PagedRequestDtoBase
    {
        /// <summary>
        /// Gets or sets the email of the user.
        /// </summary>
        public string? Email { get; set; }

        /// <summary>
        /// Gets or sets the username of the user.
        /// </summary>
        public string? UserName { get; set; }

        /// <summary>
        /// Gets or sets the field to sort the users by.
        /// </summary>
        public ApplicationUserSortBy SortBy { get; set; }

        /// <summary>
        /// Reverses the default sort behavior of the SortBy field.
        /// </summary>
        public bool ReverseSort { get; set; }
    }
}